torch/fx/graph_module.py

```diff
@@ -563,13 +568,56 @@ def code(self) -> str:
             raise RuntimeError('Code has not been generated! Please report a bug to PyTorch')
         return self._code
 
+    def _convert_tensor_default_args(self):
+        """
+        Make it so that placeholder arguments with a `Tensor` default argument are converted
+        to have a sentinel default value, and emit a function that replaces this sentinel
+        with the original tensor value (which has been stored away as an attribute).
+        """
+        for node in self.graph.nodes:
+            if node.op == 'placeholder' and len(node.args):
+                if isinstance(node.args[0], torch.Tensor):
+                    # HACK: hoisting this logic out of Tracer. We should probably figure
+                    # out how to do this in general when someone constructs a graph with
+                    # Tensor arguments and then manually constructs a GraphModule from it.
+                    tensor_val = node.args[0]
+                    i = 0
+                    while True:
+                        qualname = f'_tensor_constant{i}'
+                        if not hasattr(self, qualname):
+                            break
+                        i += 1
+                    setattr(self, qualname, tensor_val)
+                    with self.graph.inserting_before(node):
+                        node.args = (self.graph.get_attr(qualname),)
+
+                if isinstance(node.args[0], torch.fx.Node):
+                    default_value = node.args[0]
+                    node.args = (_DefaultTensorSentinel,)
+                    with self.graph.inserting_after(node):
+                        assert len(_null_coalesce_fn)
+                        actual_value = self.graph.call_function(_null_coalesce_fn[0])
+                    node.replace_all_uses_with(actual_value)
+                    # Populate arg now because replace_all_uses_with would interfere if we did it during
+                    # node construction
+                    actual_value.args = (node, default_value)
+
     @compatibility(is_backward_compatible=True)
-    def recompile(self) -> PythonCode:
+    def recompile(self, convert_tensor_default_args=True) -> PythonCode:
         """
         Recompile this GraphModule from its ``graph`` attribute. This should be
         called after editing the contained ``graph``, otherwise the generated
         code of this ``GraphModule`` will be out of date.
+
+        `convert_tensor_default_args` specifies whether `placeholder` nodes with default
```

Comment:
Is there any reason this isn't just always on?

Reply:
Because we're stable release and behavior change like this should be opt-out. I'm not super concerned because these default params basically didn't work before, but the option should still be available to opt out.

Reply:
I'm just wary of new arguments :P